src/text_embedding.rs

```diff
@@ -273,15 +274,18 @@ impl TextEmbedding {
                     "input_ids" => Value::from_array(inputs_ids_array)?,
                     "attention_mask" => Value::from_array(attention_mask_array)?,
                 ]?;
+
                 if self.need_token_type_ids {
-                    session_inputs
-                        .insert("token_type_ids", Value::from_array(token_type_ids_array)?);
+                    session_inputs.push((
+                        "token_type_ids".into(),
+                        Value::from_array(token_type_ids_array)?.into(),
+                    ));
                 }
 
                 let outputs = self.session.run(session_inputs)?;
 
                 // Extract and normalize embeddings
-                let output_data = outputs["last_hidden_state"].extract_tensor::<f32>()?;
+                let output_data = outputs["last_hidden_state"].try_extract_tensor::<f32>()?;
 
                 let embeddings: Vec<Vec<f32>> = output_data
                     .view()
```

Comment:
```suggestion
```
It should work fine too. Since `try_extract_tensor()`  now returns the same type.

Reply:
Same for `reranking.rs`.